app/models/coach.rb

```diff
@@ -23,7 +23,7 @@
 #
 class Coach < ApplicationRecord
   belongs_to :team
-  belongs_to :player, optional: true
+  belongs_to :associated_player, class_name: 'Player', foreign_key: 'associated_player_id', optional: true
```

Comment:
Bueno -- I knew I was doing something incorrect here but it was small potatoes at the moment.  Thanks for that cleanup!